--- a/cmake/external/flatbuffers/include/flatbuffers/base.h
+++ b/cmake/external/flatbuffers/include/flatbuffers/base.h
@@ -1,4 +1,4 @@
-#ifndef FLATBUFFERS_BASE_H_
+#if !defined(FLATBUFFERS_BASE_H_)
 #define FLATBUFFERS_BASE_H_

 // clang-format off
@@ -167,7 +167,7 @@

 #include <clocale>

-#if defined(__unix__) && !defined(__ANDROID__)
+#if defined(__GLIBC__) && !defined(__ANDROID__)
 #include <xlocale.h>
 #endif

